.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,135 @@
+name: Benchmark
+
+on:
+  pull_request_review:
+    types: [ submitted ]
+
+jobs:
+  build_modified_ckb:
+    name: Build Modified CKB
+    runs-on: [ self-hosted, Linux, X64 ]
+    if: |
+      contains(github.event.review.body, '#benchmark') &&
+      contains(fromJson('[ "janx", "doitian", "quake", "xxuejie", "zhangsoledad", "jjyr", "TheWaWaR", "driftluo", "keroro520", "yangby-cryptape", "liya2017" ]'), github.actor)
```

Comment:
### Is there a way to use a group name?

All actions have a same condition, and the condition has a lot of names, and we have to update those name in different action scripts every time when folks in or folks out.

#### An ugly alternative if no better way to check users permissions

Put the condition in a bash script, then
- Read group from a file, so we could only update one file.
- Or fetch group via GitHub APIs with a token, so we don't have to update any files.

Reply:
> Is there a way to use a group name?

There is no "group name", just Github user name.

> Read group from a file, so we could only update one file.

There are only [very limited functions we can call in `if`](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#functions). And cannot call shell scripts or read files

Reply:
The information that we can get: https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context

The functions that we can use: https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#functions

it's very limited

Reply:
If someone can do this, I think we could do this too.
An example of checking user permissions: [Has Permission Action](https://github.com/marketplace/actions/has-permission).

I didn't mean put everything in `if` statement, you can put that in a step.

Reply:
Yeah, this can be done in `steps`.

But when the process moves forward into `steps`, the job is already triggered:

  - it has entered the self-hosted runner
  - the job has started
  - if you want to exit the current job, you have to select an exit status, succeeded, failed, or canceled

As a result, I think filtering should occur at the `if` phase. It's ugly, but it's safe.

Reply:
PS: once a job is started, it will act as a Github check run for the target PR.

Reply:
> Yeah, this can be done in `steps`.
> 
> But when the process moves forward into `steps`, the job is already triggered:

A bit lack of imagination. :dog:

A pseudo workflow (without tests):
- Run a GitHub-hosted runner to check user permissions.
  - If passed, call the action in a self-hosted runner. 
  - if failed, do nothing, just returns done.

Ref: [workflow_dispatch](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_dispatch)

Reply:
Oh, it seems that `workflow_dispatch` can do some combination. That's really cool! I didn't think of it.
Thanks. I marked it and will do some experiments when I have time.

Reply:
I think `if` accepts variables, so we can define the name list in a variable.

Reply:
declaring variables is not supported